modules/siddhi-extensions/map/src/main/java/org/wso2/siddhi/extension/map/CreateFunctionExtension.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.siddhi.extension.map;
+
+import org.wso2.siddhi.core.config.ExecutionPlanContext;
+import org.wso2.siddhi.core.exception.ExecutionPlanRuntimeException;
+import org.wso2.siddhi.core.executor.ExpressionExecutor;
+import org.wso2.siddhi.core.executor.function.FunctionExecutor;
+import org.wso2.siddhi.query.api.definition.Attribute;
+import org.wso2.siddhi.query.api.exception.ExecutionPlanValidationException;
+
+import java.util.HashMap;
+
+/**
+ * create(key1,value1,key2,value2,.....,keyN,valueN) or create()
+ * Returns the created hashmap
+ * Accept Type(s): () , (Object,Object,....,Object,Object)
+ * Return Type(s): HashMap
+ */
+public class CreateFunctionExtension extends FunctionExecutor {
+
+    Attribute.Type returnType = Attribute.Type.STRING;
+    HashMap<Object, Object> hashMap = new HashMap<Object, Object>();
+
+    @Override
+    protected void init(ExpressionExecutor[] attributeExpressionExecutors, ExecutionPlanContext executionPlanContext) {
+        if ((attributeExpressionExecutors.length) % 2 == 1) {
+            throw new ExecutionPlanValidationException("Invalid no of arguments passed to map:create() function, " +
+                    "required 0 or multiple of 2, but found " + attributeExpressionExecutors.length);
+        }
+    }
+
+    @Override
+    protected Object execute(Object[] data) {
+        if (data == null) {
+            throw new ExecutionPlanRuntimeException("Data can not be null.");
+        }
+        if (data.length < 2) {
+            throw new ExecutionPlanRuntimeException("Number of values for data should be a multiple of 2");
+        }
```

Comment:
Data should be greater than two or multiplication of 2? Have to change either condition or error message.